temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -104,8 +109,6 @@ public boolean start() {
               options.getTaskExecutorThreadPoolSize(),
               workerMetricsScope,
               true);
-      Semaphore workflowTaskExecutorSemaphore =
-          new Semaphore(options.getTaskExecutorThreadPoolSize());
```

Comment:
Can you confirm that `start` is never called more than once on this class? (i.e. no restarting that may cause issues with reusing the semaphore)

Reply:
Yeah, it's protected by an AtomicBoolean in Worker. Start on the internal workers is always called once.